lms/static/scripts/frontend_apps/components/dashboard/OrderableActivityTable.tsx

```diff
@@ -0,0 +1,79 @@
+import type { DataTableProps, Order } from '@hypothesis/frontend-shared';
+import { DataTable } from '@hypothesis/frontend-shared';
+import { useOrderedRows } from '@hypothesis/frontend-shared';
+import type { OrderDirection } from '@hypothesis/frontend-shared/lib/types';
+import { useMemo, useState } from 'preact/hooks';
+
+import type { BaseDashboardStats } from '../../api-types';
+
+export type OrderableActivityTableProps<T extends BaseDashboardStats> = Pick<
+  DataTableProps<T>,
+  'emptyMessage' | 'rows' | 'renderItem' | 'loading' | 'title'
+> & {
+  columnNames: Partial<Record<keyof T, string>>;
+  defaultOrderField: keyof T;
+};
+
+/**
+ * List of columns which should start sorted in descending order
+ */
+const descendingOrderColumns: ReadonlyArray<string> = [
+  'last_activity',
+  'annotations',
+  'replies',
+];
+
+/**
+ * DataTable with ordering capabilities enabled by default, intended to show
+ * BaseDashboardStats rows
+ */
```

Comment:
I was thinking of something higher level like "Annotation activity table for dashboard views. Includes built-in support for sorting columns."